iceoryx_dds/include/iceoryx_dds/dds/cyclone_data_reader.hpp

```diff
@@ -33,28 +33,68 @@ namespace dds
 class CycloneDataReader : public DataReader
 {
   public:
+    /// @brief delete default ctor
     CycloneDataReader() = delete;
+
+    /// @brief Constructor to set cyclone data reader object
+    /// @param[in] serviceId ID of the service
+    /// @param[in] instanceId ID of the instance of the service
+    /// @param[in] eventId ID of the event
     CycloneDataReader(const capro::IdString_t serviceId,
                       const capro::IdString_t instanceId,
                       const capro::IdString_t eventId) noexcept;
+
+    /// @brief Virtual destructor
     virtual ~CycloneDataReader();
 
+    /// @brief copy ctor is explicitly deleted
     CycloneDataReader(const CycloneDataReader&) = delete;
+
+    /// @brief copy assignment is explicitly deleted
     CycloneDataReader& operator=(const CycloneDataReader&) = delete;
+
+    /// @brief move ctor is explicitly deleted
     CycloneDataReader(CycloneDataReader&&) = delete;
+
+    /// @brief move assignment is explicitly deleted
     CycloneDataReader& operator=(CycloneDataReader&&) = delete;
 
+    /// @brief Connect cylcone data reader to the underlying DDS network
     void connect() noexcept override;
 
+    /// @brief Get the size of the next sample if one is available
+    /// @return an optional pointer to the size of the next sample if one is available, cxx::nullopt_t if value is not
+    /// initialized
     iox::cxx::optional<uint32_t> peekNextSize() override;
+
+    /// @brief Check if new samples are ready to take
+    /// @return bool, true if length of samples is not null, else false
     bool hasSamples() override;
+
+    /// @brief Take the next available sample from the DDS data space.
+    /// @param[in] buffer Receive buffer in which sample will be stored.
+    /// @param[in] bufferSize Size of the provided buffer.
+    /// @return Error if unsuccessful.
```

Comment:
Maybe document the error better. Is it an error when there are no samples available or is it only an error when there is not enough space in `buffer`?